docs/Hooks.md

```diff
@@ -284,6 +284,18 @@ fastify.addHook('onRoute', (routeOptions) => {
   routeOptions.prefix
 })
 ```
+
+If you are authoring a plugin and you need to customize application routes, like modifying the options or adding new route hooks, this is the right place.
+
+```js
+fastify.addHook('onRoute', (routeOptions) => {
+  routeOptions.preSerialization = function (request, reply, payload, done) {
```

Comment:
I would warn the user that `routeOptions.preSerialization` is an array or `undefined`, so we should `push` the function in the array

Reply:
I updated the example to support both cases. Let me know if it works now.